client/src/components/HomePage/modal/JoinModal/index.jsx

```diff
@@ -17,8 +18,22 @@ class Join extends Component {
         firstName, lastName, email, password,
       } = context.data;
       if (firstName && lastName && email && password) {
-        this.setState({ signingUp: false });
-        context.updateState({ completeJoin: true });
+        if (!validator.isEmail(email)) {
+          this.setState({ signingUp: false });
```

Comment:
@HemaSAli 
It looks like there a few common lines of code that could be refactored into a function, to reduce some duplication:  The setState which sets signingUp to false and the setPopupMessage could turn into something like `this.showSignupError('please enter a valid email address!')`

Reply:
@tomduggan85 
Thanks Tom, you can see now..
I have created a function which is like this 
```
  setPopUp = (message, title, context) => {
    context.setPopUpMessage({
      message, title,
    });
  }
```

Reply:
also I improved it to be like this
```
  setPopUp = (message, title, context, signUp) => {
    context.setPopUpMessage({
      message, title,
    });
      this.setState({ signingUp: false });
  }
```
So I can send setState in it

Reply:
Cool, thanks!